### PR TITLE
Add check version wasm

### LINKED ISSUE #994: Enforce version check on wasm lib
when used used as dynamic lib.
Follow up from https://github.com/CosmWasm/wasmd/pull/988#discussion_r965976607

Would a `make build-linux` target with static compilation be a sufficient solution?

### PR DESCRIPTION
**Pull request by GNaD13**
Resolves #994 

Add check version wasmvm
(ref with [stargaze](https://github.com/public-awesome/stargaze))

### CODE REVIEW

File: x/wasm/module.go [hunk 1]
```diff
@@ -250,3 +256,50 @@ func ReadWasmConfig(opts servertypes.AppOptions) (types.WasmConfig, error) {
...
+func getExpectedLibwasmVersion() string {
+	buildInfo, ok := debug.ReadBuildInfo()
+	if !ok {
+		panic("can't read build info")
+	}
+	for _, d := range buildInfo.Deps {
+		if d.Path != "github.com/CosmWasm/wasmvm" {
+			continue
+		}
+		if d.Replace != nil {
```
Comment (alpe) [inline x/wasm/module.go:269]: excellent 🦅

**Comment by GNaD13** [comment]
> Excellent work. trophy On nit if you want to address it but not blocking.
> 
> This is very helpful and should go into v0.29.0 . I tested it manually with docker and got a nice expected `Error: libwasmversion mismatch. got: 1.1.1; expected: v1.1.0`.

Hi sir @alpe.Thanks for your review.

File: x/wasm/module.go [hunk 2]
```diff
@@ -250,3 +256,50 @@ func ReadWasmConfig(opts servertypes.AppOptions) (types.WasmConfig, error) {
...
+			return fmt.Sprintf(d.Replace.Version)
+		}
+		return fmt.Sprintf(d.Version)
```
[the remark this answers is not in the document]
Comment (GNaD13) [reply x/wasm/module.go]: I have edited it

File: x/wasm/module.go (hunk 1 above)
Comment (GNaD13) [reply x/wasm/module.go:269]: Thank you!!!

File: x/wasm/module.go [hunk 3]
```diff
@@ -217,6 +221,8 @@ func AddModuleInitFlags(startCmd *cobra.Command) {
 	startCmd.Flags().Uint32(flagWasmMemoryCacheSize, defaults.MemoryCacheSize, "Sets the size in MiB (NOT bytes) of an in-memory cache for Wasm modules. Set to 0 to disable.")
 	startCmd.Flags().Uint64(flagWasmQueryGasLimit, defaults.SmartQueryGasLimit, "Set the max gas that can be spent on executing a query with a Wasm contract")
 	startCmd.Flags().String(flagWasmSimulationGasLimit, "", "Set the max gas that can be spent when executing a simulation TX")
+
+	startCmd.PreRunE = chainPreRuns(checkLibwasmVersion, startCmd.PreRunE)
```
[the remark this answers is not in the document]
Comment (GNaD13) [reply x/wasm/module.go:225]: Thank you!!!

**Comment by webmaster128** [comment]
Nice one!

**Pull request merged**